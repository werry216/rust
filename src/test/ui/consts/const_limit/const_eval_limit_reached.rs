// only-x86_64
// check-pass
#![feature(const_eval_limit)]
#![const_eval_limit="2"]

const CONSTANT: usize = limit();
//~^ WARNING Constant evaluating a complex constant, this might take some time

fn main() {
    assert_eq!(CONSTANT, 1764);
}

const fn limit() -> usize {
    let x = 42;

    x * 42
}
