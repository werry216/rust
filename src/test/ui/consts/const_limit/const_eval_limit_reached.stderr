warning: Constant evaluating a complex constant, this might take some time
  --> $DIR/const_eval_limit_reached.rs:6:1
   |
LL | const CONSTANT: usize = limit();
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

